contrib/babelfishpg_tsql/src/pl_comp.c

```diff
@@ -1605,9 +1604,7 @@ pltsql_post_expand_star(ParseState *pstate, ColumnRef *cref, List *l)
 				/*
 				 * We found the original name; rewrite it as bbf_original_name
 				 */
-				bbf_original_name = &optstr[18];
-				bbf_original_name[strlen(te->resname)] = '\0';
-				te->resname = pstrdup(bbf_original_name);
+				te->resname = pnstrdup((char *) &optstr[18], strlen(te->resname));
```

Comment:
Just trying to understand: why would valgrind complain here?

Reply:
It's writing one byte passed the length. If bbf_original_name = palloc(10), then line 1609 does a bbf_original_name[10] = '0';